docs/src/pages/components/data-grid/filtering/filtering.md

```diff
@@ -4,162 +4,240 @@ title: Data Grid - Filtering
+};
+```
+
+In the example below, the rows with `rating > 4` and `isAdmin = true` will be displayed.
+
+```ts
+const filterModel: GridFilterModel = {
+  items: [
+    { id: 1, columnField: 'rating', operatorValue: '>', value: '4' },
+    { id: 2, columnField: 'isAdmin', operatorValue: 'is', value: 'true' },
+  ],
+  linkOperator: GridLinkOperator.And,
+};
+```
+
+If no `linkOperator` is provided, the grid will use `GridLinkOperator.Or` by default.
+
+### Initialize the filters
+
+To initialize the filters without controlling them, provide the model to the `initialState` prop.
 
 ```jsx
 <DataGrid
-  filterModel={{
-    items: [{ columnField: 'commodity', operatorValue: 'contains', value: 'rice' }],
+  initialState={{
+    filter: {
+      filterModel: {
+        items: [
+          { columnField: 'commodity', operatorValue: 'contains', value: 'rice' },
+        ],
+      },
+    },
   }}
 />
 ```
 
-{{"demo": "pages/components/data-grid/filtering/BasicFilteringGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+{{"demo": "pages/components/data-grid/filtering/InitialFilters.js", "bg": "inline", "defaultCodeOpen": false}}
 
-## Predefined filters
+### Controlled filters
 
-A filter is composed of three parts: the column to filter, the value to look for, and an operator (e.g. _contains_, _is before_, _is after_, etc.).
-On the `DataGridPro`, the `id` field is required on `filterModel.items` when multiple filters are used.
-The operator determines if a candidate value should be considered as a result.
-The candidate value used by the operator is the one corresponding to the `field` attribute or the `valueGetter` of the `GridColDef`.
-As part of the predefined column types, a set of operators is available.
-You can find the supported column types in the [columns section](/components/data-grid/columns/#column-types).
+Use the `filterModel` prop to control the filter applied on the rows.
 
-**Note**: The [`valueFormatter`](/components/data-grid/columns/#value-formatter) is only used for rendering purposes.
+You can use the `onFilterModelChange` prop to listen to changes to the filters and update the prop accordingly.
 
-The following demo allows to explore the different operators available:
+```jsx
+<DataGrid
+  filterModel={{
+    items: [{ columnField: 'commodity', operatorValue: 'contains', value: 'rice' }],
+  }}
+/>
+```
 
-{{"demo": "pages/components/data-grid/filtering/FilterOperators.js", "bg": "inline", "defaultCodeOpen": false}}
+{{"demo": "pages/components/data-grid/filtering/ControlledFilters.js", "bg": "inline", "defaultCodeOpen": false}}
 
-## Disable filtering
+## Disable the filters
 
-**Globally**
+### For all columns
 
 Filters are enabled by default, but you can easily disable this feature by setting the `disableColumnFilter` prop.
+The filters will still
 
 ```jsx
 <DataGrid disableColumnFilter />
 ```
 
-**Per column**
+{{"demo": "pages/components/data-grid/filtering/DisableFilteringGridAllColumns.js", "bg": "inline"}}
```

Comment:
@oliviertassinari do we have a way of hiding the demo preview ?
I'm not finding any in the `markdown/loader.js` of the core.

Is there a doc somewhere of the options we can pass to the `.js` files in the markdown ? 
If not it would be an interesting addition for us.

Reply:
You can add the option `"defaultCodeOpen": false` to do not show the preview.

I agree that a file to explain all the option available is a nice to have